### ISSUE TITLE
Order validation fails on payment when order contains no shippable items and shipping method selection is required.

### ISSUE THREAD

**Issue by javangriff** [craftcms/commerce#2204]
There seems to be a conflict between the `requireShippingMethodSelectionAtCheckout` config setting and orders that only contain items that are not shippable eg. orders containing only a donation.  When the below validation is run as part of the `actionPay` method on the payments controller, it should probably also check to see if `$order->availableShippingMethods` contains any results before this validation is run.

```
// Does the order require shipping
if ($plugin->getSettings()->requireShippingMethodSelectionAtCheckout && !$order->getShippingMethod()) {
  ...
}
```

- Craft version: 3.6.16
- Commerce version: 3.3.3
- PHP version: 7.2

**Comment by lukeholder** [comment]
if there is no available shipping methods, that could also mean that the order is just not stoppable to that location based on the shipping rules, but the setting could still require that they have selected a shipping method.

I think it would be better to check the order, and if every item in the order is marked as non shippable, then the setting would not apply. Would that make sense?

**Comment by javangriff** [comment]
Yeah that sounds like a good approach and would work for my particular use case. The store I am working on also offers physical products for sale alongside donations, so checking every item in the order to ensure they are all non-shippable seems like a more robust solution that allows existing shipping rule validation to still apply.

**Comment by lukeholder** [comment]
This has been fixed for then next 3.4 beta release.
